java-frontend/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -771,41 +806,34 @@ private void buildTryStatement(TryStatementTree tryStatementTree) {
     }
     Block finallyOrEndBlock = currentBlock;
     Block beforeFinally = createBlock(currentBlock);
-    List<Block> catches = new ArrayList<>();
+    Map<Type, Block> catches = new HashMap<>();
+    TryStatement tryStatement = new TryStatement();
+    tryStatement.successorBlock = finallyOrEndBlock;
+    enclosingTry.push(tryStatement);
+    enclosedByCatch.push(false);
     for (CatchTree catchTree : tryStatementTree.catches()) {
       currentBlock = createBlock(finallyOrEndBlock);
+      currentBlock.isCatchBlock = true;
+      enclosedByCatch.push(true);
       build(catchTree.block());
-      catches.add(currentBlock);
+      enclosedByCatch.pop();
+      catches.put(catchTree.parameter().type().symbolType(), currentBlock);
```

Comment:
The `catches` map can be removed in profit of directly calling `tryStatement.addCatch(Type, block)`